packages/packages/store/src/index.ts

```diff
@@ -0,0 +1,112 @@
+import {
+	createSlice,
+	ReducersMapObject,
+	configureStore,
+	combineReducers,
+	Middleware,
+	Store,
+	Slice,
+	AnyAction,
+	CreateSliceOptions,
+} from '@reduxjs/toolkit';
+
+export type {
+	Slice,
+	CreateSliceOptions,
+	PayloadAction,
+	Store,
+	Dispatch,
+	AnyAction,
+	Action,
+	MiddlewareAPI,
+	Middleware,
+} from '@reduxjs/toolkit';
+
+export { useSelector, useDispatch, Provider as StoreProvider } from 'react-redux';
+
+/**
+ * Usage:
+ *
+ * const mySlice = addSlice( ... );
+ *
+ * type MySliceState = SliceState<typeof mySlice>;
+ *
+ * const value = useSelector( ( state: MySliceState ) => state.mySlice.value );
+ */
+export type SliceState<S extends Slice> = {
+	[ key in S['name'] ]: ReturnType<S['getInitialState']>;
+}
+
+interface SlicesMap {
+	[key: Slice['name']]: Slice;
+}
+
+let instance: Store | null = null;
+let slices: SlicesMap = {};
+const pendingActions: AnyAction[] = [];
+const middlewares = new Set<Middleware>();
+
+const getReducers = () => {
+	const reducers = Object.entries( slices ).reduce( ( reducersData: ReducersMapObject, [ name, slice ] ) => {
+		reducersData[ name ] = slice.reducer;
+
+		return reducersData;
+	}, {} );
+
+	return combineReducers( reducers );
+};
+
+export const addSlice = ( ( sliceConfig: CreateSliceOptions ) => {
+	const slice = createSlice( sliceConfig );
+
+	if ( slices[ slice.name ] ) {
+		throw new Error( `Slice with name "${ slice.name }" already exists.` );
+	}
+
+	slices[ slice.name ] = slice;
+
+	return slice;
+} ) as typeof createSlice;
+
+export const addMiddleware = ( middleware: Middleware ) => {
+	middlewares.add( middleware );
+};
+
+export const dispatch = ( action: AnyAction ) => {
+	if ( ! instance ) {
+		pendingActions.push( action );
+
+		return;
+	}
+
+	return instance.dispatch( action );
+};
+
+export const createStore = () => {
+	if ( instance ) {
+		throw new Error( 'The store instance already exists.' );
+	}
+
+	instance = configureStore( {
+		reducer: getReducers(),
+		middleware: Array.from( middlewares ),
+	} );
+
+	if ( pendingActions.length ) {
+		pendingActions.forEach( ( action ) => dispatch( action ) );
+		pendingActions.length = 0;
+	}
+
+	return instance;
+};
+
+export const getStore = () => {
+	return instance;
+};
+
+export const deleteStore = () => {
+	instance = null;
+	slices = {};
+	pendingActions.length = 0;
+	middlewares.clear();
+};
```

Comment:
so you don't need to `replaceReducer( null )`?

Reply:
If I will use it, it will re-render all `useSelector` usages and everyone will get an error that their state is missing.